lib/ai_bot/entry_point.rb

```diff
@@ -56,6 +56,29 @@ def self.map_bot_model_to_user_id(model_name)
         end
       end
 
+      def self.ai_share_error(topic, guardian)
```

Comment:
I feel like this would work much better in a guardian extension...something like 

```ruby
def can_share_ai_conversation?(target)
  // checks go here
end
```

Otherwise this code is a bit weird and not very standard with other things we do. Chat uses a guardian extension here:

https://github.com/discourse/discourse/blob/446e4620a8373396f3979819ad9c7b1e45e5e802/plugins/chat/plugin.rb#L62-L67

https://github.com/discourse/discourse/blob/446e4620a8373396f3979819ad9c7b1e45e5e802/plugins/chat/lib/chat/guardian_extensions.rb